langlib/lang.xml/src/main/java/org/ballerinalang/langlib/xml/CreateElement.java

```diff
@@ -38,19 +43,53 @@
 //        isPublic = true
 //)
 public class CreateElement {
+    private static final String XML = "xml";
+    private static final String XML_NS_URI_PREFIX = "{" + XMLConstants.XML_NS_URI + "}";
+    private static final String XMLNS_NS_URI_PREFIX = "{" + XMLConstants.XMLNS_ATTRIBUTE_NS_URI + "}";
 
-    public static BXml createElement(BString name, BXml children) {
-        BXmlQName xmlqName = ValueCreator.createXmlQName(name);
+    public static BXml createElement(BString name, BMap<BString, BString> attributes, BXml children) {
+        String prefix = getPrefix(name.getValue(), attributes);
+        BXmlQName xmlqName;
+        if (prefix == "") {
+            xmlqName = ValueCreator.createXmlQName(name);
+        } else {
+            xmlqName = ValueCreator.createXmlQName(name, prefix);
+        }
         String temp = null;
         BXml xmlElement = XmlFactory.createXMLElement(xmlqName, temp);
+        xmlElement.setAttributes(attributes);
         xmlElement.setChildren(getChildren(children));
         return xmlElement;
     }
 
+    private static String getPrefix(String name, BMap<BString, BString> attributes) {
+        int parenEndIndex = name.indexOf('}');
```

Comment:
Let's change these to curly bracket and let's use the `lastIndexOf` too.

Reply:
will update.